packages/database/src/modelClasses/UserFavouriteDashboardItem.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { DatabaseModel } from '../DatabaseModel';
+import { DatabaseType } from '../DatabaseType';
+import { TYPES } from '../types';
+
+class UserFavouriteDashboardItemType extends DatabaseType {
+  static databaseType = TYPES.USER_FAVOURITE_DASHBOARD_ITEM;
+}
+
+export class UserFavouriteDashboardItemModel extends DatabaseModel {
+  get DatabaseTypeClass() {
+    return UserFavouriteDashboardItemType;
+  }
+
+  async updateRecord({ changeType, record }) {
+    switch (changeType) {
+      case 'create':
+        return this.findOrCreate(record);
+      case 'delete':
+        return this.delete(record);
+      default:
+        throw new Error(`Non supported change type: ${changeType}`);
+    }
+  }
```

Comment:
Thoughts on making this function use terminology that's a bit closer to the business terminology. eg.

```suggestion
  async favourite(userId, dashboardItemId) {
    return this.findOrCreate({dashboard_item_id: dashboardItemId, user_id: userId });
  }
  
  async unfavourite(userId, dashboardItemId) {
    return this.delete({dashboard_item_id: dashboardItemId, user_id: userId });
  }
```

Then we can just do:
`models.userFavouriteDashboardItem.favourite(userId, dashboardItemId);`
and
`models.userFavouriteDashboardItem.unfavourite(userId, dashboardItemId);`